yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/request/AbstractMultiLookupRequest.java

```diff
@@ -0,0 +1,211 @@
+package tech.ytsaurus.client.request;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import tech.ytsaurus.client.ApiServiceUtil;
+import tech.ytsaurus.client.rpc.RpcClientRequestBuilder;
+import tech.ytsaurus.core.YtTimestamp;
+import tech.ytsaurus.rpc.TRequestHeader;
+import tech.ytsaurus.rpcproxy.TReqMultiLookup;
+
+/**
+ * Base class for multi lookup requests.
+ */
+public abstract class AbstractMultiLookupRequest<
+        TBuilder extends AbstractMultiLookupRequest.Builder<TBuilder, TRequest>,
+        TRequest extends AbstractMultiLookupRequest<TBuilder, TRequest>> extends RequestBase<TBuilder, TRequest> {
+
+    @Nullable
+    protected final YtTimestamp timestamp;
+    @Nullable
+    protected final YtTimestamp retentionTimestamp;
+    @Nullable
+    protected final ReplicaConsistency replicaConsistency;
+    protected final List<LookupRowsRequest> subrequests;
```

Comment:
I suggest to create class MultiLookupSubrequest (analogue of [this](https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/client/api/dynamic_table_client.h#L39)).